### DISCUSSION TITLE
Using in a non-monorepo on Vercel

### DISCUSSION THREAD

**Discussion by JamesSingleton** [vercel/turborepo#6408] [Help] [answered]
I am looking into this in order to reduce some of my build times on Vercel (roughly 20 minutes for 340 static pages). I noticed the `non-monorepo` example says to run `pnpm turbo build` but I'm using `npm`. So I looked at the `with-npm` example and the build script is `turbo run build`. However, if I update mine to that with a `turbo.json` that looks like

```json
{
  "$schema": "https://turbo.build/schema.json",
  "globalDependencies": ["**/.env.*local"],
  "pipeline": {
    "build": {
      "outputs": [".next/**", "!.next/cache/**"]
    },
    "lint": {},
    "dev": {
      "cache": false,
      "persistent": true
    }
  }
}
```

I will get

```
• Running build
• Remote caching disabled
root task build (turbo run build) looks like it invokes turbo and might cause a loop

No tasks were executed as part of this run.

 Tasks:    0 successful, 0 total
Cached:    0 cached, 0 total
  Time:    885ms

 ERROR  run failed: command  exited (1)
```

So I'm not sure what the _correct_ way to set that part up. Also, on Vercel does that mean I have to override the build command to something else?

**Comment by anthonyshew** [comment, marked as answer]
You will want to use the `non-monorepo` example as your start and then run `npx @turbo/workspaces convert` to easily switch to a different package manager.
